include/roboteam_ai/stp/PlayEvaluator.h

```diff
@@ -0,0 +1,121 @@
+//
+// Created by maxl on 18-02-21.
+//
+
+#ifndef RTT_PLAYEVALUATOR_H
+#define RTT_PLAYEVALUATOR_H
+
+#include "world/World.hpp"
+
+namespace rtt::ai::stp{
+    enum class GlobalEvaluation{
+        /// Game States
+        BallPlacementThemGameState,
+        BallPlacementUsGameState,
+        FreeKickThemGameState,
+        FreeKickUsGameState,
+        HaltGameState,
+        KickOffThemGameState,
+        KickOffThemPrepareGameState,
+        KickOffUsGameState,
+        KickOffUsPrepareGameState,
+        NormalOrFreeKickUsGameState,
+        NormalPlayGameState,
+        PenaltyThemGameState,
+        PenaltyThemPrepareGameState,
+        PenaltyUsGameState,
+        PenaltyUsPrepareGameState,
+        StopGameState,
+        /// Global Evaluations
+        BallCloseToThem,
+        BallCloseToUs,
+        BallClosestToUs,
+        BallGotShot,
+        BallIsFree,
+        BallMovesSlow,
+        BallOnOurSide,
+        BallOnTheirSide,
+        BallShotOrCloseToThem,
+        DistanceFromBall,
+        FreedomOfRobots,
+        GoalVisionFromBall,
+        GoalVision,
+        NoGoalVisionFromBall,
+        WeHaveBall,
+        WeHaveMajority
+    };
+
+    class PlayEvaluator {
+    public:
+        struct PlayScoring {
+            uint8_t evaluationScore;
+            double weight;
+        };
+
+        /**
+         * Gets the score of a Global Evaluation, if it was not updated yet, update it before.
+         * @param evaluation that needs
+         * @return
+         */
+        uint8_t getGlobalEvaluation(GlobalEvaluation evaluation);
+
+        /**
+         * Sets this->world
+         * @param world World to update against
+         */
+        void update(world::World* world) noexcept;
+
+        /**
+         * Gets this world, used when a play is not initialised yet (in between plays)
+         * @return world
+         */
+        world::World* getWorld() noexcept;
```

Comment:
Why does the playevaluator need to provide others with the world? This seems like an architectural mistake to me. You should be getting the world from world.

Reply:
My thought was:
In between plays PlayEvaluator is active. In one tick the world gets updated in there, and only in that tick is used to determine stuff with. getworld from PlayEvaluator solves the issue that only the current play (that will be replaced) has the world instance, the other dont have it. Therefore, to determine the score from the game state, a world has to be given to them. My thinking was to use the PlayEvaluator for that.

How would i improve this?